client-js/src/client/errors.js

```diff
@@ -0,0 +1,203 @@
+ * with a response with `5xx` status code.
+ *
+ * Indicates an unhandled exception was thrown upon the request processing.
+ *
+ * @extends ServerError
+ */
+export class InternalServerError extends ServerError {
+
+  /**
+   * @param {!Response} response the server response caused this error
+   */
+  constructor(response) {
+    super(response.statusText, response);
+  }
+}
+
+/**
+ * An error which occurs when sending off a request to Spine server endpoint results with a
+ * malformed response which can't be processed.
+ *
+ * @extends ServerError
+ */
+export class ResponseProcessingError extends ServerError {
+
+  /**
+   * @param {?Error} cause the reason why this error occurred
+   */
+  constructor(cause) {
+    super(cause.message || cause, cause);
+  }
+}
+
+/**
+ * An error which occurs when sending off a request to Spine server endpoint results
+ * with a response with `4xx` status code.
+ *
+ * @extends ClientError
+ */
+export class RequestProcessingError extends ClientError {
+
+  /**
+   * @param {!Response} response the server response caused this error
+   */
+  constructor(response) {
+    super(response.statusText, response);
+  }
+}
+
+/**
+ * An error which occurs when sending off a command to Spine server endpoint results with a response
+ * containing {@code spine.base.Ack} with error status. It means that a command message was rejected further
+ * processing (e.g. because of a validation error).
+ *
+ * @extends ClientError
+ */
+export class CommandProcessingError extends ClientError {
+  /**
+   * @param {spine.base.Error} error the technical error occurred upon receiving the request and
+   *                                  no further processing would occur.
+   */
+  constructor(error) {
+    super(error.message, error);
+  }
+
+  /**
+   * Returns the type of {@code spine.base.Error}.
+   *
+   * @return {string}
+   */
+  type() {
+    return this.getCause().type;
+  }
+
+  /**
+   * Returns the code of {@code spine.base.Error}.
+   *
+   * @return {number}
+   */
+  code() {
+    return this.getCause().code;
+  }
+
+  /**
+   * Returns an optional validation error object error indicating that a
+   * message sent to the server did not pass validation.
+   *
+   * @return {?spine.validate.ValidationError}
+   */
+  validationError() {
+    return this.getCause().validationError;
+  }
+}
+
+/**
+ * Typedef representing the type union of `client-js` module errors.
+ *
+ * @typedef {Object} Errors
+ *
+ * @property {SpineWebError} SpineWebError
+ * @property {ConnectionError} ConnectionError
+ * @property {ClientError} ClientError
+ * @property {ServerError} ServerError
+ * @property {RequestProcessingError} RequestProcessingError
+ * @property {CommandProcessingError} CommandProcessingError
+ * @property {ResponseProcessingError} ResponseProcessingError
+ * @property {InternalServerError} InternalServerError
+ */
+
+/**
+ * The object which represents the type union of `client-js` module errors.
```

Comment:
The npm modules name is `spine-web-client`.
```suggestion
 * All of Spine Web Client errors gathered in a single namespace. 
 * 
 * These errors can be checked using `instanceof`, but beware the order of subsequent checks, because the errors are arranged in a hierarchy.
 * 
 * Base error class for all errors thrown by `spine-web-client` is `SpineError`.
```